conda/plugins/manager.py

```diff
@@ -476,6 +487,33 @@ def load_settings(self) -> None:
         for name, (parameter, aliases) in self.get_settings().items():
             add_plugin_setting(name, parameter, aliases)
 
+    def get_environment_specifiers(self, filename: str) -> CondaEnvironmentSpecifier:
+        """
+        Returns the environment_spec plugin that can handle the provided file.
+
+        Raises PluginError if more than one environment_spec plugin is found to be able to handle the file.
+        Raises EnvironmentSpecPluginNotDetected if no plugins were found.
+        """
+        hooks = self.get_hook_results("environment_specifiers")
+        found = []
+        for hook in hooks:
+            if hook.environment_spec(filename).can_handle():
+                found.append(hook)
+
+        if len(found) == 1:
+            return found[0]
+        elif len(found) > 0:
+            # raise an error if there is more than one plugin found
+            names = ", ".join([hook.name for hook in found])
+            raise PluginError(
+                f"Multiple environment_spec plugins registered for the environment specified in '{filename}'. Please review plugins: {names}"
```

Comment:
took a stab at writing a helpful message here. I feel like there are better words to use, but I can't find them. Feel free to push directly to my branch or suggest a better message!